File: FITCCRS2App.Models/Models/Projekat.cs
using System;
namespace FITCCRS2App.Models.Models
{
	public class Projekat
	{

        public int ProjekatId { get; set; }
        public string Naziv { get; set; } = null!;
        public string Opis { get; set; } = null!;
        public int KategorijaId { get; set; }
        public int TimId { get; set; }

        public virtual Kategorija Kategorija { get; set; } = null!;
        public virtual Tim Tim { get; set; } = null!;
        //public virtual ICollection<Rezultat> Rezultats { get; set; }
    }
}

File: FITCC.UI/admin_fitcc/lib/models/projekat.dart

import 'package:admin_fitcc/models/kategorija.dart';
import 'package:admin_fitcc/models/tim.dart';
import 'package:json_annotation/json_annotation.dart';

part 'projekat.g.dart';

@JsonSerializable(explicitToJson: true)
class Projekat {
    int projekatId;
    String naziv;
    String opis;
    int kategorijaId;
    int timId;
    Kategorija kategorija; 
    Tim tim; 

  Projekat(
     this.projekatId,
     this.naziv,
     this.opis,
     this.kategorijaId,
     this.timId,
     this.kategorija,
     this.tim,  );

  factory Projekat.fromJson(Map<String, dynamic> json) => _$ProjekatFromJson(json);
  Map<String, dynamic> toJson() => _$ProjekatToJson(this);
}

